### PR TITLE
Add titlecheck

### PR DESCRIPTION
**Pull request by jandersson**
Adds a script checking recently added titles.

This is a nice script if you are curious about new titles but don't want to read through them to notice any new ones. 

e.g. Do I have any new titles?
```
R>title list moonmage                                                                                                                                                                                                           
The following titles of the Moonmage category are available to you:                                                                                                                                                             
                                                                                                                                                                                                                                
Initiate Moon Mage, Apprentice Moon Mage, Journeyman Moon Mage, Moon Mage, Expert Moon Mage, 
Master Moon Mage, Grand Master Moon Mage, Legendary Moon Mage, Transcendent Moon Mage, Stargazer, 
Medium, Contemplator, Mentalist, Mesmerist, Dreamer, Lunar Mage, Lunar Magician, Ascendant, Dreamweaver, 
Illusionist, Novus, Occultist, Spectromancer, Puppet Master, Manipulator, Auspex, Enthraller, 
Astronomer, Lunarian, Wild Card, Astrologist, Lightcrafter, Luminary, Scion of Yavash, Scion of Xibar, 
Scion of Katamba, Illuminated, Lucky, Palm Reader, Austromancer, Chiromancer, Soothsayer, 
Luckbringer, Daydreamer, Augur, Fortuneteller, Visionist, Doomsayer, Dowser, Mystic, Predictor, Determinist, Diviner, Mantic, Seer, Fatebound, Aura Reader, Prophet, Omen, Ill Omen, 
Forecaster, Prescient, Omen Speaker, Forerunner, Portent, Portent Seeker, Far Seer, Precursor, 
Harbinger, Fateshaper, Fateshackled, Alda Pelan, Pupil, Cold Reader, Meteorologist, Docent, Mathematician, Extrapolator, Philosopher, Geometer, Logarithmancer, Numinist, Sage, Logician, 
Fatalist, Cosmologist, Enlightened Geometer, Prognosticator, Alda Remoren, Duskwalker, Darkling, 
Fade, Shadow of Fate, Goldcap Aspirant, Observer, Clairaudient, Clairvoyant, Discerner, 
Moromancer, Charlatan, Pseudomancer, Dissident, Mystic Blade, Fate Blade, Psychokinetic, 
Telekinetic Warrior, Lightwielder, Mindflayer, Steelstorm, Shadow Reaver, Light Weaver, 
Shadow Mage, Sliver Slinger, Hex, Displacer, Fate Reaver, Telekinetic Fury, Battlefield Scribe, 
Battlefield Scholar, Stalwart Scholar, Mystic Aegis, Geometric Warder, Spellbreaker, Spellbane, 
Astral Guardian, Nemesis, Prophetic Warrior, Precognitive Soldier, Y'Shai Aspirant, Heritage Keeper
, Archivist, Proctor, Gestalt, Accord Keeper, Bibliomancer, Stoichomancer, Traveler, Hypnotist, Psionic,
 Witness, Astral Traveler, All-Seeing, Untethered
```

```
s>;titlecheck moonmage                                                                                                                                                                                                       
--- Lich: titlecheck active.                                                                                                                                                                                                 
[titlecheck]>title pre list moonmage                                                                                                                                                                                                                                                                                                                                                                                          
The following titles of the Moonmage category are available to you:                                                                                                                                                          
... lots                                                                                                                                                       
-------------------                                                                                                                                                                                                          
Title Check                                                                                                                                                                                                                  
-------------------                                                                                                                                                                                                          
You have no new titles                                                                                                                                                                                                       
-------------------                                                                                                                                                                                                          
--- Lich: titlecheck has exited.

### CODE REVIEW

File: titlecheck.lic [hunk 1]
```diff
@@ -0,0 +1,66 @@
+=begin
+This script requires that you have ;dependency installed... it will not work otherwise
```
Comment (rpherbig) [inline titlecheck.lic]: 💡 Please move this documentation to elanthipedia

File: titlecheck.lic [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+Checks for new titles since the last time you ran ;titlecheck.
+Usage: ;titlecheck <title set>  (e.g. ;titlecheck moonmage)
+
+author: Crannach
+game: dr
+=end
+
+class TitleCheck
+  include DRC
```
Comment (rpherbig) [inline titlecheck.lic]: ❓ Does this need `custom_require([common])`?

File: titlecheck.lic [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  def initialize
+    arg_definitions = [
+        [
+        {name: 'title_set', options: %w(blunt ranged brawling genedged specedged thrown pole shield slings weapons weaponmaster performer primarymagic magic money ownership survival1 survival2 survival3 lore criminal generic racial premium order religion novice practitioner dilettante aficionado adept expert professional authority genius savant master grandmaster guru legend cleric trader warmage necromancer empath ranger paladin bard barbarian thief moonmage), description: 'What set of titles are you checking?'}
+      ] 
+    ]
+    args = parse_args(arg_definitions)
+    UserVars.titles = Hash.new if UserVars.titles.nil?
```
Comment (rpherbig) [inline titlecheck.lic]: 💡 I _think_ this could also be written as `UserVars.titles |= {}` if you find that clearer

File: titlecheck.lic (hunk 2 above)
Comment (Sarvatt) [reply titlecheck.lic]: yep for the bput

File: titlecheck.lic (hunk 3 above)
Comment (Sarvatt) [reply titlecheck.lic]: `UserVars.titles ||= {}` rather?

Comment (rpherbig) [reply titlecheck.lic]: Yep, good catch

**Comment by jandersson** [comment]
Updated based on feedback

**Pull request merged**